kfs-core/src/main/java/edu/arizona/kfs/pdp/batch/AchBankInputFileType.java

```diff
@@ -0,0 +1,371 @@
+		return KFSConstants.ACHFileConstants.ACH_BANK_FILE_TYPE_IDENTIFIER;
+	}
+
+	@Override
+	public String getFileName(String principalName, Object parsedFileContents, String fileUserIdentifier) {
+		String fileName = "bank_" + principalName;
+		if (StringUtils.isNotBlank(fileUserIdentifier)) {
+			fileName += "_" + fileUserIdentifier;
+		}
+		fileName += "_" + dateTimeService.toDateTimeStringForFilename(dateTimeService.getCurrentDate());
+		
+		fileName = StringUtils.remove(fileName,  " ");
+		
+		return fileName;
+	}
+
+	@Override
+	public Object parse(byte[] fileByteContent) throws ParseException {
+		List<ACHBank> bankList = new ArrayList<ACHBank>();
+		List<BankReportLine> reportLines = new ArrayList<BankReportLine>();
+		String systemSource = KFSConstants.EMPTY_STRING;
+		ACHBank currentBank;
+		ACHBank existingBank;
+		BankReportLine reportLine;
+		Boolean saveBank = true;
+		BufferedReader bufferedFileReader = new BufferedReader(new InputStreamReader(new ByteArrayInputStream(fileByteContent)));
+		String fileLine;
+		
+		try {
+			while((fileLine = bufferedFileReader.readLine()) != null) {
+				LOG.info("Bank line: " + fileLine);
+				
+				//get a valid systemSource from the file
+				if (StringUtils.isBlank(systemSource)) {
+					systemSource = fileLine.substring(150, 152).trim();
+				}
+				
+				currentBank = new ACHBank(fileLine);
+				existingBank = getAchBank(currentBank.getBankRoutingNumber());
+				if (ObjectUtils.isNull(existingBank)) {
+					saveBank = validateBank(currentBank, reportLines);
+					if (saveBank) {
+						reportLine = new BankReportLine(currentBank.getBankRoutingNumber(), currentBank.getBankName(), KFSConstants.ACHFileConstants.ACH_BANK_LOADED_SUCCESSFULLY);
+						reportLines.add(reportLine);
+						businessObjectService.save(currentBank);
+						bankList.add(currentBank);
+					}
+				}
+			}
+			
+			writeReport(reportLines, systemSource);
+		}
+		catch (IOException e) {
+			LOG.error("Error encountered reading from file content", e);
+			throw new ParseException("Error encountered reading from file content", e);
+		}
+		
+		return bankList;
+	}
+
+	@Override
+	public boolean validate(Object parsedFileContents) {
+		return true;
+	}
+
+	@Override
+	public void process(String fileName, Object parsedFileContents) {
+		//empty method so compiler doesn't complain
+	}
+
+	@Override
+	public String getAuthorPrincipalName(File file) {
+		String[] fileNameParts = StringUtils.split(file.getName(), "_");
+		if (fileNameParts.length >= 2) {
+			return fileNameParts[1];
+		}
+		return null;
+	}
+
+	@Override
+	public String getTitleKey() {
+		return KFSKeyConstants.MESSAGE_BATCH_UPLOAD_TITLE_ACH_BANK;
+	}
+	
+	private ACHBank getAchBank(String bankRoutingNumber) {
+		
+		Map<String, String> pkMap = new HashMap<String, String>();
+		pkMap.put(PdpPropertyConstants.BANK_ROUTING_NUMBER, bankRoutingNumber);
+		ACHBank bank = (ACHBank) businessObjectService.findByPrimaryKey(ACHBank.class, pkMap);
```

Comment:
Unnecessary cast, a hold out when the BOS was not generisized. Since we already have to make a change to catch that reader in the other comment, we could sneak this one in too. Don't worry about squashing if so.